tests/integration/security/ca_custom_root/trust_domain_alias_secure_naming_test.go

```diff
@@ -0,0 +1,242 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cacustomroot
+
+import (
+	"fmt"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/connection"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+const (
+	HTTPS  = "https"
+	POLICY = `
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "mtls"
+spec:
+  mtls:
+    mode: STRICT
+---
+apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "server-naked"
+spec:
+  host: "*.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+`
+)
+
+// TestTrustDomainAliasSecureNaming scope:
+// The client side mTLS connection should validate the trust domain alias during secure naming validation.
+//
+// Setup:
+// 1. Setup Istio with custom CA cert, to make sure all workloads have same root cert because of #3
+// 2. One client workload with istio-proxy sidecar injected
+// 3. Two naked server workloads with custom certs in two different trust domain reflected in the SAN (spiffe)
+// 4. PeerAuthentication with strict mtls, to enforce the mtls connection
+// 5. DestinaitonRule with tls ISTIO_MUTUAL mode, because Istio auto mTLS will let client sends plaintext to naked servers by default
+// 6. MeshConfig.TrustDomainAliases contains one of the trust domain in #3
+//
+// Expectation:
+// 1. When the trust domain of server TLS certificate is in the list of MeshConfig.TrustDomainAliases,
```

Comment:
could be simplified by: client requests to "server-naked-foo" succeeds, and requests to "server-naked-bar" fails.

Or if you want to keep current. 2) can be removed by: "Otherwise, requests fail."